common/src/main/java/io/netty/util/internal/PlatformDependent0.java

```diff
@@ -722,6 +721,11 @@ static void putShort(long address, short value) {
         UNSAFE.putShort(address, value);
     }
 
+    static void putShortOrdered(long adddress, short newValue) {
+        UNSAFE.storeFence();
+        UNSAFE.putShort(null, adddress, newValue);
```

Comment:
Maybe we can also use a varhandle and see if UNSAFE has the "correct" method first and if this is the case directly use it and if not use the workaround here

Reply:
We can use varhandler, but varhandler was introduced in Java 9. https://openjdk.org/jeps/193

Maybe we can modify the code here after we raise the baseline version to 11?